source/opt/ir_context.cpp

```diff
@@ -183,5 +188,248 @@ void IRContext::KillNamesAndDecorates(Instruction* inst) {
+        SpvOpInBoundsAccessChain,
+        SpvOpArrayLength,
+        SpvOpVectorExtractDynamic,
+        SpvOpVectorInsertDynamic,
+        SpvOpVectorShuffle,
+        SpvOpCompositeConstruct,
+        SpvOpCompositeExtract,
+        SpvOpCompositeInsert,
+        SpvOpCopyObject,
+        SpvOpTranspose,
+        SpvOpSampledImage,
+        SpvOpImageSampleImplicitLod,
+        SpvOpImageSampleExplicitLod,
+        SpvOpImageSampleDrefImplicitLod,
+        SpvOpImageSampleDrefExplicitLod,
+        SpvOpImageSampleProjImplicitLod,
+        SpvOpImageSampleProjExplicitLod,
+        SpvOpImageSampleProjDrefImplicitLod,
+        SpvOpImageSampleProjDrefExplicitLod,
+        SpvOpImageFetch,
+        SpvOpImageGather,
+        SpvOpImageDrefGather,
+        SpvOpImageRead,
+        SpvOpImage,
+        SpvOpConvertFToU,
+        SpvOpConvertFToS,
+        SpvOpConvertSToF,
+        SpvOpConvertUToF,
+        SpvOpUConvert,
+        SpvOpSConvert,
+        SpvOpFConvert,
+        SpvOpQuantizeToF16,
+        SpvOpBitcast,
+        SpvOpSNegate,
+        SpvOpFNegate,
+        SpvOpIAdd,
+        SpvOpFAdd,
+        SpvOpISub,
+        SpvOpFSub,
+        SpvOpIMul,
+        SpvOpFMul,
+        SpvOpUDiv,
+        SpvOpSDiv,
+        SpvOpFDiv,
+        SpvOpUMod,
+        SpvOpSRem,
+        SpvOpSMod,
+        SpvOpFRem,
+        SpvOpFMod,
+        SpvOpVectorTimesScalar,
+        SpvOpMatrixTimesScalar,
+        SpvOpVectorTimesMatrix,
+        SpvOpMatrixTimesVector,
+        SpvOpMatrixTimesMatrix,
+        SpvOpOuterProduct,
+        SpvOpDot,
+        SpvOpIAddCarry,
+        SpvOpISubBorrow,
+        SpvOpUMulExtended,
+        SpvOpSMulExtended,
+        SpvOpAny,
+        SpvOpAll,
+        SpvOpIsNan,
+        SpvOpIsInf,
+        SpvOpLogicalEqual,
+        SpvOpLogicalNotEqual,
+        SpvOpLogicalOr,
+        SpvOpLogicalAnd,
+        SpvOpLogicalNot,
+        SpvOpSelect,
+        SpvOpIEqual,
+        SpvOpINotEqual,
+        SpvOpUGreaterThan,
+        SpvOpSGreaterThan,
+        SpvOpUGreaterThanEqual,
+        SpvOpSGreaterThanEqual,
+        SpvOpULessThan,
+        SpvOpSLessThan,
+        SpvOpULessThanEqual,
+        SpvOpSLessThanEqual,
+        SpvOpFOrdEqual,
+        SpvOpFUnordEqual,
+        SpvOpFOrdNotEqual,
+        SpvOpFUnordNotEqual,
+        SpvOpFOrdLessThan,
+        SpvOpFUnordLessThan,
+        SpvOpFOrdGreaterThan,
+        SpvOpFUnordGreaterThan,
+        SpvOpFOrdLessThanEqual,
+        SpvOpFUnordLessThanEqual,
+        SpvOpFOrdGreaterThanEqual,
+        SpvOpFUnordGreaterThanEqual,
+        SpvOpShiftRightLogical,
+        SpvOpShiftRightArithmetic,
+        SpvOpShiftLeftLogical,
+        SpvOpBitwiseOr,
+        SpvOpBitwiseXor,
+        SpvOpBitwiseAnd,
+        SpvOpNot,
+        SpvOpBitFieldInsert,
+        SpvOpBitFieldSExtract,
+        SpvOpBitFieldUExtract,
+        SpvOpBitReverse,
+        SpvOpBitCount,
+        SpvOpDPdx,
+        SpvOpDPdy,
+        SpvOpFwidth,
+        SpvOpDPdxFine,
```

Comment:
Something I learned today:  The derivative instructions exchange information with other invocations (in the same quad), so they should not be considered a combinator.  However, this commit is mainly a refactor, so I'll file a separate issue to fix that later.